apis/python/src/tiledbsoma/io/registration/ambient_label_mappings.py

```diff
@@ -0,0 +1,402 @@
+            var_axes=var_axes,
+        )
+
+    @classmethod
+    def from_isolated_anndata(
+        cls,
+        adata: ad.AnnData,
+        *,
+        measurement_name: str,
+        obs_field_name: Optional[str] = None,
+        var_field_name: Optional[str] = None,
+    ) -> Self:
+        """Factory method to compute an label-to-SOMA-join-ID mappings for a single input file in
+        isolation. This is used when a user is ingesting a single AnnData/H5AD to a single SOMA
+        experiment, not in append mode, but allowing us to still have the bulk of the ingestor code
+        to be non-duplicated between non-append mode and append mode.
+        """
+        tiledbsoma.logging.logger.info("Registration: registering AnnData object.")
+
+        obs_axis = AxisAmbientLabelMapping.from_isolated_dataframe(
+            adata.obs, index_field_name=obs_field_name
+        )
+        tiledbsoma.logging.logger.info(
+            f"Registration: accumulated to nobs={len(obs_axis.data)}"
+        )
+
+        var_axes = {}
+        var_axis = AxisAmbientLabelMapping.from_isolated_dataframe(
+            adata.var,
+            index_field_name=var_field_name,
+        )
+        var_axes[measurement_name] = var_axis
+        tiledbsoma.logging.logger.info(
+            f"Registration: accumulated to nvar={len(var_axis.data)}"
+        )
+        if adata.raw is not None:
+            raw_var_axis = AxisAmbientLabelMapping.from_isolated_dataframe(
+                adata.raw.var,
+                index_field_name=var_field_name,
+            )
+            var_axes["raw"] = raw_var_axis
+            tiledbsoma.logging.logger.info(
+                f"Registration: accumulated to nvar={len(raw_var_axis.data)}"
+            )
+
+        return cls(
+            obs_axis=obs_axis,
+            var_axes=var_axes,
+        )
+
+    @classmethod
+    def from_isolated_h5ad(
+        cls,
+        h5ad_file_name: str,
+        *,
+        measurement_name: str,
+        obs_field_name: Optional[str] = None,
+        var_field_name: Optional[str] = None,
+    ) -> Self:
+        """Factory method to compute label-to-SOMA-join-ID mappings for a single input file in
+        isolation. This is used when a user is ingesting a single AnnData/H5AD to a single SOMA
+        experiment, not in append mode, but allowing us to still have the bulk of the ingestor code
+        to be non-duplicated between non-append mode and append mode.
+        """
+        adata = ad.read_h5ad(h5ad_file_name, "r")
+        return cls.from_isolated_anndata(
+            adata,
+            measurement_name=measurement_name,
+            obs_field_name=obs_field_name,
+            var_field_name=var_field_name,
+        )
+
+    @classmethod
+    def from_isolated_soma_experiment(
+        cls,
+        experiment_uri: Optional[str] = None,
+        *,
+        obs_field_name: str = "obs_id",
+        var_field_name: str = "var_id",
+        # XXX CTX/CFG
```

Comment:
note-to-self comment from development?

Reply:
> OK, but this is a module, not a class. And given that this will be exporting public-facing APIs, and it is exporting several classes/functions, it is kind of weird to have "none of it is user-facing API" in the docstring (is it temporary in case we do a release with this included but not finished?). Whatever the goal is is fine, just the docstring and the code itself are sending mixed messages.

@thetorpedodog and I want to defer the leading-underscore, exports, etc -- visibility questions -- until a follow-on PR wherein there _is_ user-facing content (there isn't any yet) and it will be clearer what needs to happen where.

I appreciate and value your feedback. I want to action this particular feedback-point on a follow-on PR.